### PR TITLE
implement V3 types and their instances

### PR DESCRIPTION
**Pull request by chfanghr**
This is the most boring code I have ever written. I fucking hate typescript.

TODOs:

- [x] Doc strings
- [x] Generators for round trip tests

### CODE REVIEW

**Comment by szg251** [comment]
@jaredponn  I wanted to bump the version to v1.1.0 (only minor bump, as we're not introducing any breaking changes).
However using it it lambda-buffers caused some build issues, it seems to be still looking for 1.0.0. Do you know how to fix it? https://hercules-ci.com/accounts/github/mlabs-haskell/derivations/%2Fnix%2Fstore%2Fxgw3vk4xjd8cc39bszwd1m07hj5x7kvy-lbf-prelude-typescript-1.0.0.drv/log?via-job=c08fd4b9-1eae-408f-b77a-3996cb816deb

**Comment by jaredponn** [comment]
Argh, I'm guessing it's because the `package-lock.json` files depend on version 1.0.0, so it's still looking for that but it doesn't exist.

To fix it, one would probably regenerate all the `package-lock.json` files in LB -- I know it's ugly.

File: Tx.ts [hunk 1]
```diff
@@ -0,0 +1,274 @@
+/**
```
Comment (jaredponn) [inline Tx.ts]: Er what is this file doing here? This file corresponds to the `V1/Tx.hs` Plutus Ledger API module which already exists in `src/Lib/V1/Tx.ts` https://github.com/mlabs-haskell/plutus-ledger-api-typescript/blob/main/src/Lib/V1/Tx.ts.

Perhaps this got duplicated somehow? I'd guess this probably isn't necessary.

File: src/Lib/V3/Tx.ts [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+      throw new JsonError(`TxId should be 32 bytes`);
+    }
+    return bs.fields;
+  },
+};
+
+/**
+ * {@link IsPlutusData} instance for {@link TxId}
+ *
+ * Note this includes the `0` tag.
```
Comment (jaredponn) [inline src/Lib/V3/Tx.ts]:
```suggestion
```

Looks like the newer representation is a bit more efficient and doesn't include the tag. See the `deriving newtype `clause of https://github.com/input-output-hk/plutus/blob/1.36.0.0/plutus-ledger-api/src/PlutusLedgerApi/V3/Tx.hs#L42-L54}

**Review by jaredponn — APPROVED** [review]
Amazing!

File: Tx.ts (hunk 1 above)
Comment (szg251) [reply Tx.ts]: oops, I just recently transitioned to a GUI IDE, and dragging files randomly :) maybe sticking to vim is not a bad idea

File: src/Lib/V3/Tx.ts (hunk 2 above)
Comment (szg251) [reply src/Lib/V3/Tx.ts]: Ah, thank you, I forgot to update the docs

**Pull request merged**